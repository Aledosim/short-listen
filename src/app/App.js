import React from 'react';
import style, { createGlobalStyle } from 'styled-components';

import TopBar from '../features/topbar/TopBar'
import TrackList from '../features/tracklist/TrackList'

// colors
const background = "#0A0A0A"

const GlobalStyle = createGlobalStyle`
	* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
	}

  body {
    background-color: ${background};
  }

	@media(max-width: 1080px) {
    html {
      font-size: 93.75%;
    }
	}

	@media(max-width: 720px) {
    html {
      font-size: 87.5%;
    }
	}

  button {
    cursor: pointer;
  }
`

const Header = style.header`
  position: fixed;
  width: 100%;
  z-index: 1;

  background-color: ${background};
`

const TrackListContainer = style.div`
  position: relative;
  top: 4.5rem;
`

function App() {
  return (
    <>
      <GlobalStyle />
      <Header>
        <TopBar />
      </Header>
      <TrackListContainer>
        <TrackList />
      </TrackListContainer>
    </>
  );
}

export default App;
