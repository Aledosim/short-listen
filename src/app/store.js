import { configureStore } from '@reduxjs/toolkit';

import trackListReducer from '../features/tracklist/trackListSlice'
import favoritesReducer from '../features/favorites/favoritesSlice'

export const store = configureStore({
  reducer: {
    trackList: trackListReducer,
    favorites: favoritesReducer,
  },
});
