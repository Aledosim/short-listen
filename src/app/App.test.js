import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from './store';
import App from './App';

function testRender() {
  return render(
    <Provider store={store}>
      <App />
    </Provider>
  )
}

it('renders without crashing', async () => {
  const tree = testRender()

  expect(tree).toMatchSnapshot()
})
