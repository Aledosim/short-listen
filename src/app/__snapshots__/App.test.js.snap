// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-height: 50px;
  padding-top: 0.75rem;
  padding-bottom: 0.5rem;
}

.c2 {
  margin-left: 2.5rem;
  max-width: 100px;
  height: auto;
}

.c3 {
  position: relative;
  margin: auto;
  width: 30rem;
  height: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  width: 100%;
  height: 80%;
  padding-right: 4rem;
  padding-left: 2rem;
  background-color: #F4F1F1;
  border: none;
  border-radius: 4px;
}

.c4::-webkit-input-placeholder {
  color: #000000,46%;
}

.c4::-moz-placeholder {
  color: #000000,46%;
}

.c4:-ms-input-placeholder {
  color: #000000,46%;
}

.c4::placeholder {
  color: #000000,46%;
}

.c5 {
  position: relative;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  right: 9px;
  height: 70%;
  width: 3rem;
  border: none;
  background-image: url(search_icon.svg);
  background-repeat: no-repeat;
  background-color: #F4F1F1;
  color: #9B9B9B;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #EBEBEB;
}

.c0 {
  position: fixed;
  width: 100%;
  z-index: 1;
  background-color: #0A0A0A;
}

.c6 {
  position: relative;
  top: 4.5rem;
}

@media (max-width:1080px) {

}

@media (max-width:720px) {

}

<body>
    <div>
      <header
        class="c0"
      >
        <header
          class="c1"
          data-cy="topbar"
        >
          <svg
            class="c2"
          >
            deezer_logo.svg
          </svg>
          <form
            class="c3"
          >
            <input
              autocomplete="on"
              autosave="true"
              class="c4"
              data-cy="searchField"
              id="searchField"
              placeholder="Album, artist or track title"
              type="search"
            />
            <button
              class="c5"
              data-cy="searchButton"
              type="submit"
            />
          </form>
        </header>
      </header>
      <div
        class="c6"
      >
        <div
          role="list"
        >
          <div
            class="infinite-scroll-component__outerdiv"
          >
            <div
              class="infinite-scroll-component c7"
              style="height: auto; overflow: auto;"
            >
              <h4>
                Loading...
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-dIsUp bQxFdd"
    >
      <header
        class="sc-bdnxRM looujR"
        data-cy="topbar"
      >
        <svg
          class="sc-gtsrHT cBzwgK"
        >
          deezer_logo.svg
        </svg>
        <form
          class="sc-dlnjwi jYCUyy"
        >
          <input
            autocomplete="on"
            autosave="true"
            class="sc-hKFxyN cdSgUH"
            data-cy="searchField"
            id="searchField"
            placeholder="Album, artist or track title"
            type="search"
          />
          <button
            class="sc-eCApnc kAxBQA"
            data-cy="searchButton"
            type="submit"
          />
        </form>
      </header>
    </header>
    <div
      class="sc-bqGGPW rovGd"
    >
      <div
        role="list"
      >
        <div
          class="infinite-scroll-component__outerdiv"
        >
          <div
            class="infinite-scroll-component sc-crzoAE biPuQf"
            style="height: auto; overflow: auto;"
          >
            <h4>
              Loading...
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
